stream-legal-entity/legal-entity-core/src/main/java/com/backbase/stream/ServiceAgreementSagaV2.java

```diff
@@ -176,12 +176,20 @@ private Mono<ServiceAgreementTaskV2> postServiceAgreementContacts(ServiceAgreeme
         } else {
             externalUserId = externalUserOptional.get();
         }
-        return contactsSaga.executeTask(createContactsTask(streamTask.getId(), serviceAgreement.getExternalId(),
-                serviceAgreement.getExternalId(), externalUserId, AccessContextScope.SA, serviceAgreement.getContacts()))
+        return contactsSaga.executeTask(
+                createContactsTask(streamTask.getId(), getLegalEntityBySharingAccounts(serviceAgreement),
```

Comment:
does it mean that instead of the service agreement external id you have to pass the legal entity external id here?

Reply:
the SA_ID is the third parameter. it was wrongly called with SA_ID as LE_ID in the second parameter and it is always failing with the old implementation. unless, accidentally SA_ID and LE_ID are equal.